### ISSUE TITLE
Songs not Playing

### ISSUE THREAD

**Issue by goodmorty** [ampache/ampache#1127]
There are some songs that would formerly play but now will not play in anything, not on web or subsonic. It isn't a file thing as I haven't done anything to the song files themselves in a long while.

**Comment by goodmorty** [comment]
Some of said songs will play after hitting stop and then hitting play in the web player

**Comment by Phyks** [comment]
Same problem here, although it is not only happening with FLAC songs. Some songs crash the web player and I have to pass them manually.

However, in my particular case, it seems to happen *only* in "random song" mode. If I play these songs manually or by full album, I do not have such problems.

P.S. : Maybe linked to https://github.com/ampache/ampache/issues/1128 as well.

**Comment by Niols** [comment]
Being administrator of the server @Phyks is using, I've put the logs in verbose (5) level.
Surprisingly, **nothing** appears. As if the song weren't even in the playlist. Maybe a problem with the client?

**Comment by Phyks** [comment]
Here is a screenshot of the player when it is happening. As you can see, it is in an incoherent state showing the previous music instead of this one, although it is correctly hilighted in the right panel.

![2016-02-10-122859](https://cloud.githubusercontent.com/assets/3856586/12946058/f171f224-cff1-11e5-935c-ae24060e145b.png)

I do not have any error in the JS console. In the network debugging, I can see a successful AJAX request (status 200) on `ajax.server.php?page=index&action=reloadnp`. Response content seems to be ok, although I did not have a deep look at it.

There are not much more infos available on the client side :/

Most of the songs having the issue in my case seem to be MP3 VBR. Playing the music directly rather than through random songs playlist works fine.

**Comment by Afterster** [comment]
I cannot see the reason for now. Which 'random' feature are you talking about (there is a few :smile:)? Thank you.

**Comment by Phyks** [comment]
I am talking about "Random > Track" feature in the left panel.

**Comment by Afterster** [comment]
Cannot reproduce. Are you all using Javascript player tech here? Or flash, or html5?
If you don't know please try to disable `Authorize JavaScript decoder (Aurora.js) in Web Player(s)` on Preferences => Streaming.

**Comment by Phyks** [comment]
Left only the HTML5 player in the settings, and can reproduce. Used a vanilla Chromium to confirm it is not coming from my browser. @Niols was not able to reproduce either though. :/

**Comment by dmersiowsky** [comment]
I also have this problem (Web Player, HTML5, also Lullaby for Android). There also is a strange workaround: Whenever I start to download a file, I will be able to stream for a couple of minutes, even other songs than the one I downloaded.

Also, when not doing the workaround, the requests to the actual mp3 seem to get stuck on server side.

**Comment by ciroiriarte** [comment]
Not sure if this is related, but currently I cannot play some mp3 files. Going to the Chrome debug console I see this:

![image](https://cloud.githubusercontent.com/assets/1750260/13433594/46d1b442-dfb1-11e5-8a37-d4e89b4d4b66.png)

The file can be found in the path stated on the song detail, it's owned by the apache user and I can copy it to my desktop and hear it without issues.

**Comment by dmersiowsky** [comment]
Did you try to re-index your library? It might not be in sync with the actual mp3 files.

**Comment by dmersiowsky** [comment]
Oh, and regarding my post above: Updating solved the issue for me. HTTPS is still an issue, but I think this is offtopic here.

**Comment by ciroiriarte** [comment]
Yup, run many times a catalog update. It didn't change anything.

**Comment by Niols** [comment]
I reproduced the issue. But I have nothing more to say than what has already been said.

Re-indexing the library isn't the problem, since it is done every night, and since I can listen to the song when taken individually.

**Comment by goodmorty** [comment]
Almost every song requires that I hit play again or stop then play now, so somehow it's gotten worse.

**Comment by Phyks** [comment]
Same thing for me, but it comes from me having disabled autoplay in Firefox (so it was expected and my fault). Ensure you did not do anything like this.

**Comment by Phyks** [comment]
Do others manage to reproduce the issue? I reproduced it with both Chromium and Firefox, but have no idea where it comes from. I'll keep having a look at it to see if I can find anything.

**Comment by Phyks** [comment]
Ok, I am working on it at the moment.

First, when viewing a song which cannot be played, no network request are made, which is weird. (On below screenshot, the 3 requests are from the previously played song).
![2016-06-16-175304](https://cloud.githubusercontent.com/assets/3856586/16123968/467711b4-33ed-11e6-8051-fba6e98b9760.png)

Then, after some tests, it seems it happens only with flac files.

JPlayer does not seem to support Flac files and they should be played back using Aurora.JS.

* If I play a single Flac song, everything is working fine as expected ("Unsupported media type" unless I enable Aurora.JS, and in this case the song is played).
* If I play a playlist with a Flac song and a MP3 song, it is not working as described in previous comments, whatever the configuration of Flash/HTML/Aurora player used. The behaviour is the same across all possible configurations.

I am trying to understand better what is happening when there is both a MP3 file and a FLAC file.

**Comment by Phyks** [comment]
I reproduced it in a minimal jPlayer demo file (`blue.monday/demo-02.html` slightly modified. It seems to match exactly [this issue](https://github.com/happyworm/jPlayer/issues/281).

Code used is:
```
$(document).ready(function(){

	new jPlayerPlaylist({
		jPlayer: "#jquery_jplayer_1",
		cssSelectorAncestor: "#jp_container_1"
	}, [
		{
			title:"Cro Magnon Man",
			mp3:"http://www.jplayer.org/audio/mp3/TSP-01-Cro_magnon_man.mp3",
		},
		{
			title:"Test flac",
			flac:"file:///tmp/Vampire%20Weekend%20-%206%20-%20Run.flac",
		}
	], {
		swfPath: "../../dist/jplayer",
        solution: "html, aurora",
		supplied: "mp3,flac",
		wmode: "window",
		useStateClassSkin: true,
		autoBlur: false,
		smoothPlayBar: true,
		keyEnabled: true
	});
});
```

**Comment by Phyks** [comment]
"Waiting" label as this seems to be an error on jPlayer/Aurora side. Discussion should move to the above mentionned issue I guess.

**Comment by Phyks** [comment]
Still, this is a really annoying bug. What about moving "aurora" to the leading positions in "solution" array, which is a kind of ugly, but temporary patch?

**Comment by Phyks** [comment]
Also, should we try to solve the issue in jPlayer or consider moving to another player like VideoJS?

**Comment by Phyks** [comment]
Patch submitted upstream: https://github.com/happyworm/jPlayer/pull/365.

**Comment by Phyks** [comment]
Up? It looks like the PR in jPlayer is not going to be merged anytime soon, and then, the issue will remain for quite a long time in Ampache.

Plus, I am trying to figure out an alternative interface and am looking at audio players in the browser at the moment.

So far, I can see 2 possibilities:
1. Either jPlayer is the best one, and I really think we should fork it, review the pending PRs on the upstream project and backport those interesting in our fork, and maintain it any longer. At least if jPlayer is, as it seems, completely abandoned.
2. Consider switching to an alternative player if jPlayer is not state of the art. But I have only seen video.js which could withstand the comparison, and I do not have any expertise and overview on it to have a strong idea about which is the best.

**Comment by Afterster** [comment]
For me, jPlayer is abandoned. See https://groups.google.com/forum/#!searchin/jplayer/abandoned%7Csort:date/jplayer/XzsibB5QLz0/uH-90RhlAQAJ
I understand they don't have time (I personally don't have much this year because of my work) but they didn't try to found someone to help them.

1) Forking the project is a lot of work to do it properly and to keep it alive. It could be done to match ampache goals only but will still requires some work. Would be better if someone else than current Ampache developers would like to do it so we can keep our efforts on Ampache. We can try to find such guys on the frustrated jPlayer community maybe.
2) I also did some trial on this way, mainly with videojs (as you may have noticed on other threads / my videojs plugin repositories). The results months ago was that it wasn't ready for 'music only' streams/playlists but this could have changed. Otherwise it will also requires some huge work, to improve videojs to be a better music only player, and to reintegrate this player inside Ampache. There is also some good HTML5 only music player but I found it limiting (no good video feature support, html5 only)

**Comment by Phyks** [comment]
OK =(

Personnally, I don't have any contacts in the jPlayer or videoJS community.

As this issue is really a blocking one for many users, I would personnally tend to fork jPlayer under Ampache namespace, market it explicitly as being forked and maintained for the sole purpose of Ampache, and integrate the fixes. Then, this "urgent" issue will be solved, and we will have some relief to discuss videojs and others solutions more in depth.

What do you think?

**Comment by Afterster** [comment]
I believe that as long Ampache depends on jPlayer, this would be the correct way to do it. Maybe we will go back to original repository if maintainers continue their work one day...

I'm just not pushing it as I will not be the one to do the work here for the moment :sweat_smile:.
I hope we will get some contributors from jPlayer community otherwise the fork will not be successful in the long term. So basicaly, I agree with you :smile:. Do you have permissions to do so on Ampache organization belief or should I?

**Comment by Phyks** [comment]
I cannot fork under Ampache organization. And even if I could, I would not do it until you told me it was ok :)

So we can:
1. Fork it under Ampache namespace, edit the README to state this is an Ampache maintained fork, providing support and maintenance only for Ampache purpose.
2. I will push a PR with my fix for this issue in the fork. I will push a PR to track this fork with the fix in Ampache.
(At this point, this issue will be solved)

As a more longterm solution, we should all discuss about best solution to implement (which will require some amount of work). Either stick with jPlayer (then review the current PRs in jPlayer, merge those which are important, have a look at current issues), or move to something else (which one? video.js?).

From what I have seen so far, the only true alternative (if not html5 only) seems to be video.js. Solutions I see so far are:
1. Stick with jPlayer, and maintain it.
2. Move to video.js, but you were telling it was not perfect.
3. Move to an html5 only player, but this means we may have a *lot* of transcoding to do on the fly (everything which is not Aurora.JS supported nor HTML5 supported).

@ampache/developers Do any other dev have an opinion on this?

**Comment by Phyks** [comment]
So, I am starting to have a deeper look at it as I am going to add a webplayer to my alternative interface.

So far, if I think of the major music formats that we should support (I probably forgot some), I can think of:

| Formats | Aurora | HTML5                       | Flash    |
| ------- | ------ | --------------------------- | -------- |
| MP3     | Yes    | Yes                         | Yes      |
| OGG     | Yes    | Yes, -IE, -Safari           |          |
| WAV     |        | Yes, -IE                    |          |
| FLAC    | Yes    |                             |          |
| AAC     | Yes    | Yes, -Firefox               | Yes (?)* |
| ALAC    | Yes    |                             |          |

\* To be tested, not explicit in https://helpx.adobe.com/flash/kb/supported-codecs-flash-player.html

_Note_: "-Safari" includes iOS devices as well.


Some extra notes:
* All major browsers available today support the `<audio>` tag, so Flash fallback should be almost useless.
* Aurora + HTML5 is a valid solution to play almost all the files we have. But it has several issues. First, seeking in aurora is not always implemented (in FLAC files for instance). Then, `AudioWorker`s are not yet available and it could not offload decoding to another thread, resulting in laggy audio when there is too much load.

Things to consider:
* First solution is to stick with jPlayer and Aurora. jPlayer code is getting quite old and is no longer supported. This would mean forking jPlayer and merging PRs (having a quick look at them looks like it is not a really difficult task) and trying to solve Aurora.JS issues.
* The only other reliable solutions I found so far are [video.js](http://videojs.com/) but mainly oriented towards video and [howler.js](https://github.com/goldfire/howler.js) which is maintained but rely only on the browser support.
* Another approach would be to transcode more things to play in the browser and [this](https://github.com/goldfire/howler.js/#format-recommendations) might be interesting.

**Comment by nikhilroy2** [comment]
same problem

**Comment by AshotN** [comment]
FLAC is now standard on modern browsers. @nikhilroy2 if you are having a similar issue and it hasn't been resolved, please open a new issue.

**Comment by lachlan-00** [comment]
This is now fixed in all branches. This was a jplayer issue / chrome issue by the looks d37bda7d4
